OrbitQt/topdownwidget.cpp

```diff
@@ -136,3 +169,23 @@ void TopDownWidget::onCustomContextMenuRequested(const QPoint& point) {
     ui_->topDownTreeView->collapseAll();
   }
 }
+
+void TopDownWidget::on_searchLineEdit_textEdited(const QString& text) {
+  if (text.isEmpty()) {
+    return;
+  }
+  auto* proxy_model =
+      static_cast<QSortFilterProxyModel*>(ui_->topDownTreeView->model());
```

Comment:
This feels reverse and is also potentially dangerous. I would suggest either to move the QSortFilterProxyModel into TopDownWidget or to move this method to where QSortFilterProxyModel lives. TopDownWidget could have a Qt-signal "filter_changed(const QString&)" and you can connect that signal to a function or lambda that has proper access to the proxy model.

Reply:
I'm not sure I understood. Could you elaborate? Why do you think it's dangerous and why does it feel reverse? Are you suggesting on keeping the `QSortFilterProxyModel` as a field of `TopDownWidget` and why is it necessary when it's stored in the `QTreeView`, or did I just get it all wrong?

Reply:
Ahh. Never mind. It already "lives" in TopDownWidget. So this is fine.

Nit: I would prefer to access the proxy model via a member variable in TopDownWidget, so bascially having the pointer to the proxy model in TopDownWidget like you suggested.

Thanks and sorry for the delay!

Reply:
Done, I had actually done it originally and then for some reason I changed my mind.